cmd/ttn-lw-cli/commands/end_device_templates.go

```diff
@@ -0,0 +1,451 @@
+	"github.com/spf13/pflag"
+	"go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/internal/api"
+	"go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/internal/io"
+	"go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/internal/util"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/pkg/types"
+)
+
+var (
+	endDeviceTemplateFlattenPaths = []string{"end_device.provisioning_data"}
+)
+
+func templateFormatIDFlags() *pflag.FlagSet {
+	flagSet := &pflag.FlagSet{}
+	flagSet.String("format-id", "", "")
+	return flagSet
+}
+
+var (
+	errNoTemplateFormatID             = errors.DefineInvalidArgument("no_template_format_id", "no template format ID set")
+	errEndDeviceMappingNotFound       = errors.DefineNotFound("mapped_end_device_not_found", "end device mapping not found")
+	errNoEndDeviceTemplateJoinEUI     = errors.DefineInvalidArgument("no_end_device_template_join_eui", "no end device template JoinEUI set")
+	errNoEndDeviceTemplateStartDevEUI = errors.DefineInvalidArgument("no_end_device_template_start_dev_eui", "no end device template start DevEUI set")
+)
+
+func getTemplateFormatID(flagSet *pflag.FlagSet, args []string) string {
+	var formatID string
+	if len(args) > 0 {
+		if len(args) > 1 {
+			logger.Warn("multiple IDs found in arguments, considering only the first")
+		}
+		formatID = args[0]
+	} else {
+		formatID, _ = flagSet.GetString("format-id")
+	}
+	if formatID == "" {
+		return ""
+	}
+	return formatID
+}
+
+var (
+	endDeviceTemplatesCommand = &cobra.Command{
+		Use:     "templates",
+		Aliases: []string{"template", "tmpl"},
+		Short:   "End Device template commands",
+	}
+	endDeviceTemplatesCreateCommand = &cobra.Command{
+		Use:               "create [flags]",
+		Short:             "Create an end device template with mapping key",
+		PersistentPreRunE: preRun(),
+		RunE: asBulk(func(cmd *cobra.Command, args []string) error {
+			forwardDeprecatedDeviceFlags(cmd.Flags())
+			paths := util.UpdateFieldMask(cmd.Flags(), setEndDeviceFlags, attributesFlags())
+
+			var res ttnpb.EndDeviceTemplate
+			if inputDecoder != nil {
+				_, err := inputDecoder.Decode(&res)
+				if err != nil {
+					return err
+				}
+				paths = append(paths, res.FieldMask.Paths...)
+			}
+
+			if mappingKey, _ := cmd.Flags().GetString("mapping-key"); mappingKey != "" {
+				res.MappingKey = mappingKey
+			}
+			if err := util.SetFields(&res.EndDevice, setEndDeviceFlags); err != nil {
+				return err
+			}
+			res.EndDevice.Attributes = mergeAttributes(res.EndDevice.Attributes, cmd.Flags())
+			res.FieldMask.Paths = ttnpb.BottomLevelFields(paths)
+
+			return io.Write(os.Stdout, config.OutputFormat, &res)
+		}),
+	}
+	endDeviceTemplatesFromDeviceCommand = &cobra.Command{
+		Use:   "from-device [flags]",
+		Short: "Create an end device template from an existing device",
+		Long: `Create an end device template from an existing device
+
+	This command takes end devices from stdin.`,
```

Comment:
I think we should also start documenting what the flags do (they select fields in this case) and which fields are included/excluded by default.